autonomous.py

```diff
@@ -111,8 +113,37 @@ def executeAuton(self):
             inaccuracyDistance = math.sqrt(pow(relativePose.X(), 2) + pow(relativePose.Y(), 2))
             PPLibTelemetry.setPathInaccuracy(inaccuracyDistance)
 
-            #if the speeds are minimal and minimum time has elapsed, move onto the next task
-            if(abs(self.chassisSpeeds.vx/self.maxSpeed) < 0.1 and abs(self.chassisSpeeds.vy/self.maxSpeed) < 0.1 and abs(self.chassisSpeeds.omega_dps) < 5 and self.autonTimer.get() - self.lastTime > self.pathTrajectory.getTotalTimeSeconds()):
+            self.noteDriveCounter += 1
+
+            #if there is a note, it is within range, waitTime isn't negative, and the waitTime has passed, then use note detection
+            if(self.notedetector.hasTarget() and waitTime >= 0 and self.notedetector.getTargetErrorY() < self.maxPickUpDistance and self.autonTimer.get() - self.lastTime > waitTime):
+                #move every other robot cycle
+                if(self.noteDriveCounter % 2 == 0):
+                    # Use chassisSpeeds for forwards and rotational movement, but use notedetector for lateral movement
+                    self.moduleStates = self.swervometer.getKinematics().toSwerveModuleStates(ChassisSpeeds(self.chassisSpeeds[0], self.drivetrain.noteDrive_x_pid_controller.calculate(self.notedetector.getTargetErrorX()) * 4.3 * 10, self.chassisSpeeds[2]))
+                    self.modules = self.drivetrain.getModules()
+                    self.modules['front_left'].move(self.moduleStates[0].speed / (self.maxSpeed), (self.moduleStates[0].angle.degrees() + 270) % 360)
+                    self.modules['front_right'].move(self.moduleStates[1].speed / (self.maxSpeed), (self.moduleStates[1].angle.degrees() + 270) % 360)
+                    self.modules['rear_left'].move(self.moduleStates[2].speed / (self.maxSpeed), (self.moduleStates[2].angle.degrees() + 270) % 360)
+                    self.modules['rear_right'].move(self.moduleStates[3].speed / (self.maxSpeed), (self.moduleStates[3].angle.degrees() + 270) % 360)
+                    self.modules['front_left'].execute()
+                    self.modules['front_right'].execute()
+                    self.modules['rear_left'].execute()
+                    self.modules['rear_right'].execute()
+                else:
+                    self.moduleStates = self.swervometer.getKinematics().toSwerveModuleStates(ChassisSpeeds(self.chassisSpeeds[0]/2, 0, 0))
```

Comment:
Why do we halve the speed for the non-detection case now? My sense is that this is what you and Eric were discussing at the end, correct?

Reply:
We slow down in order to catch up to NetworkTables, but it could be done through other methods too